src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -118,57 +235,189 @@ MemoryInput::MemoryInput(const std::shared_ptr<ov::Node>& op, const GraphContext
+        inPortConfig.constant(false);
+        inPortConfig.setMemDesc(descCreators.at(LayoutType::ncsp)->createSharedDesc(precision, shape));
+
+        config.inConfs.push_back(std::move(inPortConfig));
+    }
+
+    PortConfig outPortConfig;
+
+    outPortConfig.inPlace(0);
+    outPortConfig.constant(false);
+    outPortConfig.setMemDesc(descCreators.at(LayoutType::ncsp)->createSharedDesc(precision, shape));
+
+    config.outConfs.push_back(std::move(outPortConfig));
+
+    supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
-/**
- * Copy data from one tensor into other.
- * As is. Assume that data is dense tensor with same layout.
- * @param dst destination memory object
- * @param src source memory object
- */
-inline
-static void simple_copy(const IMemory& dst, const IMemory& src) {
-    auto srcPtr = static_cast<uint8_t*>(src.getData());
-    auto dstPtr = static_cast<uint8_t*>(dst.getData());
-    if (src.getDataType() == dst.getDataType()) {
-        auto srcSizeInByte = src.getSize();
-        auto dstSizeInByte = dst.getSize();
+void MemoryInput::initOptimalPrimitiveDescriptor() {
+    // Mimic the child node memory desc to avoid extra reorder
+    auto childEdge = getChildEdgeAt(0);
+    auto child = childEdge->getChild();
+    auto childPd = child->getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(childPd,
+        child->getTypeStr(), " ",
+        child->getName(),
+        "failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
 
-        IE_ASSERT(srcSizeInByte == dstSizeInByte) << "MemoryNode objects are not compatible. Has different sizes.";
+    const auto& childConfig = childPd->getConfig();
+    auto mem_desc = childConfig.inConfs[childEdge->getOutputNum()].getMemDesc();
 
-        cpu_memcpy(dstPtr, srcPtr, srcSizeInByte);
-    } else {
-        cpu_convert(srcPtr, dstPtr, src.getDesc().getPrecision(),
-            dst.getDesc().getPrecision(), src.getDesc().getShape().getElementsCount());
+    auto selectedPd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selectedPd,
+        "MemoryInput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto config = selectedPd->getConfig();
+    config.outConfs.front().setMemDesc(mem_desc);
+    //bypass any checks, we enforce the child descriptor
+    selectedPd->setConfig(config);
+}
+
+void MemoryInput::resolveInPlaceEdges(Edge::LOOK look) {
+    if (!(look & Edge::LOOK_UP)) {
+        Node::resolveInPlaceEdges(look);
+        return;
+    }
+
+    auto selected_pd = getSelectedPrimitiveDescriptor();
+    OPENVINO_ASSERT(selected_pd,
+        "MemoryInput ",
+        getName(),
+        " failed getSelectedPrimitiveDescriptor() call, preferable primitive descriptor is not set");
+
+    auto memDesc = selected_pd->getConfig().outConfs.front().getMemDesc();
+    memMngr = std::make_shared<ProxyMemoryMngr>();
```

Comment:
Can we use the same way of change_default_ptr to MemoryInput, instead of using a proxy mem manager?

Reply:
In theory yes. This is rather an implementation details hidden inside the MemoryInput node, they doesn't affect anything outside. And neither of the solution is better then the other in this case.